sunpy/data/_sample.py

```diff
@@ -77,41 +73,70 @@ def download_sample_data(progress=True, overwrite=True, timeout=None):
     -------
     None
     """
-    # Creating the directory for sample files to be downloaded
-    sampledata_dir = get_and_create_sample_dir()
-
-    number_of_files_fetched = 0
-    print("Downloading sample files to {}".format(sampledata_dir))
+    print("Downloading all sample files to {}. Overwriting if necessary.".format(sampledata_dir))
     for file_name in six.itervalues(_files):
-        if not overwrite:
-            if os.path.isfile(os.path.join(sampledata_dir,
-                                           file_name[0])):
-                number_of_files_fetched += 1
-                continue
-
-        for base_url in _base_urls:
-            full_file_name = file_name[0] + file_name[1]
+        get_sample_file(file_name, url_list=_base_urls, overwrite=True)
+
+
+def get_sample_file(filename, url_list, progress=True, overwrite=False, timeout=None):
+    """
+    Downloads a sample file. Will download  a sample data file and move it to the sample data directory.
+    Also, uncompresses zip files if necessary. Returns the local file if exists.
+
+    Parameters
+    ----------
+    filename: str
+        Name of the file
+    url_list: str or list
+        urls where to look for the file
+    progress: `bool`
+        Show a progress bar during download
+    overwrite: `bool`
+        If True download and overwrite an existing file.
+    timeout: `float`
+        The timeout in seconds. If `None` the default timeout is used from
+        `astropy.utils.data.Conf.remote_timeout`.
+
+    Returns
+    -------
+    result: str
+        The local url of the file downloaded. None if it failed.
```

Comment:
local url? are we accessing to these files via a local server? I would say: "the local path".

Also... should just quietly fail? I'm not sure whether that's good... it looks like Fortran, then we have to check whether the result is a path or not... cannot we just raise an error?